sky/provision/fluidstack/instance.py

```diff
@@ -89,18 +89,45 @@ def run_instances(region: str, cluster_name_on_cloud: str,
 
     while True:
         instances = _filter_instances(cluster_name_on_cloud, pending_status)
+        if len(instances) > config.count:
+            raise RuntimeError(
+                f'Cluster {cluster_name_on_cloud} already has '
+                f'{len(instances)} nodes, but {config.count} are '
+                'required. Please try terminate the cluster and retry.')
         if not instances:
             break
         logger.info(f'Waiting for {len(instances)} instances to be ready.')
         time.sleep(POLL_INTERVAL)
     exist_instances = _filter_instances(cluster_name_on_cloud, ['running'])
     head_instance_id = _get_head_instance_id(exist_instances)
 
+    def rename(instance_id, new_name):
+        try:
+            utils.FluidstackClient().rename(instance_id, new_name)
+        except Exception as e:
+            logger.warning(f'run_instances error: {e}')
+            raise
+
+    for inst_id, inst in exist_instances.items():
+        if head_instance_id is None:
+            head_instance_id = inst_id
+            rename(inst_id, f'{cluster_name_on_cloud}-head')
+            logger.info(f'Rename head node {head_instance_id}.')
+        if inst_id != head_instance_id and inst['hostname'].endswith('-head'):
```

Comment:
Is this for the case when launching on an existing cluster where some node has been set to head node? in this case, should we keep the original head node?

Reply:
Why would there be multiple `-head`? A bug on the CSP side? Worth a comment?